src/Xtools/ArticleInfo.php

```diff
@@ -0,0 +1,989 @@
+        $interval = date_diff($dateLast, $dateFirst, true);
+        return $interval->format('%a');
+    }
+
+    /**
+     * Get the average number of days between edits to the page.
+     * @return double
+     */
+    public function averageDaysPerEdit()
+    {
+        return round($this->getTotalDays() / $this->getNumRevisions(), 1);
+    }
+
+    /**
+     * Get the average number of edits per day to the page.
+     * @return double
+     */
+    public function editsPerDay()
+    {
+        $editsPerDay = $this->getTotalDays()
+            ? $this->getNumRevisions() / ($this->getTotalDays() / (365 / 12 / 24))
+            : 0;
+        return round($editsPerDay, 1);
+    }
+
+    /**
+     * Get the average number of edits per month to the page.
+     * @return double
+     */
+    public function editsPerMonth()
+    {
+        $editsPerMonth = $this->getTotalDays()
+            ? $this->getNumRevisions() / ($this->getTotalDays() / (365 / 12))
+            : 0;
+        return round($editsPerMonth, 1);
+    }
+
+    /**
+     * Get the average number of edits per year to the page.
+     * @return double
+     */
+    public function editsPerYear()
+    {
+        $editsPerYear = $this->getTotalDays()
+            ? $this->getNumRevisions() / ($this->getTotalDays() / 365)
+            : 0;
+        return round($editsPerYear, 1);
+    }
+
+    /**
+     * Get the average number of edits per editor.
+     * @return double
+     */
+    public function editsPerEditor()
+    {
+        return round($this->getNumRevisions() / count($this->editors), 1);
+    }
+
+    /**
+     * Get the percentage of minor edits to the page.
+     * @return double
+     */
+    public function minorPercentage()
+    {
+        return round(
+            ($this->minorCount / $this->getNumRevisions()) * 100,
+            1
+        );
+    }
+
+    /**
+     * Get the percentage of anonymous edits to the page.
+     * @return double
+     */
+    public function anonPercentage()
+    {
+        return round(
+            ($this->anonCount / $this->getNumRevisions()) * 100,
+            1
+        );
+    }
+
+    /**
+     * Get the percentage of edits made by the top 10 editors.
+     * @return double
+     */
+    public function topTenPercentage()
+    {
+        return round(($this->topTenCount / $this->getNumRevisions()) * 100, 1);
+    }
+
+    /**
+     * Get the number of times the page has been viewed in the given timeframe.
+     * @param  int $latest Last N days.
+     * @return int
+     */
+    public function getPageviews($latest)
+    {
+        return $this->page->getLastPageviews($latest);
+    }
+
+    /**
+     * Get the page assessments of the page.
+     * @see https://www.mediawiki.org/wiki/Extension:PageAssessments
+     * @return string[]
+     */
+    public function getAssessments()
+    {
+        if (!is_array($this->assessments)) {
+            $this->assessments = $this->page->getAssessments();
```

Comment:
I'm getting a Twig error when requesting `/articleinfo/en.wikisource.org/Main_Page`: `Impossible to access an attribute ("assessment") on a boolean variable ("").` because `Page::getAssessments()` can also return false.

Reply:
Yeah I think is actually breaking in production too. Might as well fix it with this PR, I'm on it!